internal/handlers/pg/pgdb/indexes.go

```diff
@@ -16,49 +16,118 @@ package pgdb
 
 import (
 	"context"
+	"errors"
 
 	"github.com/jackc/pgx/v4"
 
+	"github.com/FerretDB/FerretDB/internal/types"
+	"github.com/FerretDB/FerretDB/internal/util/iterator"
 	"github.com/FerretDB/FerretDB/internal/util/lazyerrors"
+	"github.com/FerretDB/FerretDB/internal/util/must"
 )
 
-// indexParams contains parameters for creating an index.
-// TODO This type will become exported in https://github.com/FerretDB/FerretDB/issues/1509 (similar to QueryParams).
-type indexParams struct {
-	db         string   // FerretDB database name
-	collection string   // FerretDB collection name
-	index      string   // FerretDB index name
-	key        indexKey // Index specification (pairs of field names and sort orders) // TODO
-	unique     bool     // Whether the index is unique
+// IndexParams contains parameters for creating an index.
+type IndexParams struct {
+	Index  string   // FerretDB index name
+	Key    IndexKey // Index specification (pairs of field names and sort orders) // TODO
+	Unique bool     // Whether the index is unique
 }
 
-// indexKey defines a type for index key - pairs of field names and sort orders.
-type indexKey []indexKeyPair
+// IndexKey defines a type for index key - pairs of field names and sort orders.
+type IndexKey []IndexKeyPair
 
-// indexKeyPair consists of a field name and a sort order that are part of the index.
-type indexKeyPair struct {
-	field string
-	order indexOrder
+// IndexKeyPair consists of a field name and a sort order that are part of the index.
+type IndexKeyPair struct {
+	Field string
+	Order IndexOrder
 }
 
-// indexOrder defines a type for index sort order.
-type indexOrder int8
+// IndexOrder defines a type for index sort order.
+type IndexOrder int8
 
-// indexOrder constants.
+// IndexOrder constants.
 const (
-	indexOrderAsc  indexOrder = 1
-	indexOrderDesc indexOrder = -1
+	IndexOrderAsc  IndexOrder = 1
+	IndexOrderDesc IndexOrder = -1
 )
 
+// Indexes returns a list of indexes for the given database and collection.
+//
+// If the given collection does not exist, it returns ErrTableNotExist.
+func Indexes(ctx context.Context, tx pgx.Tx, db, collection string) ([]IndexParams, error) {
+	metadata, err := newMetadata(tx, db, collection).get(ctx, false)
+	if err != nil {
+		return nil, err
+	}
+
+	if !metadata.Has("indexes") {
+		return []IndexParams{}, nil
+	}
+
+	indexes := must.NotFail(metadata.Get("indexes")).(*types.Array)
+
+	res := make([]IndexParams, indexes.Len())
+	iter := indexes.Iterator()
+
+	defer iter.Close()
+
+	for {
+		i, idx, err := iter.Next()
+
+		switch {
+		case err == nil:
+			idx := idx.(*types.Document)
+			key := must.NotFail(idx.Get("key")).(*types.Document)
+
+			res[i] = IndexParams{
+				Index:  must.NotFail(idx.Get("name")).(string),
+				Unique: must.NotFail(idx.Get("unique")).(bool),
+				Key:    make([]IndexKeyPair, 0, key.Len()),
+			}
+
+			keyIter := key.Iterator()
```

Comment:
```suggestion
			keyIter := key.Iterator()
			defer keyIter.Close()
```
Just for the consistency.